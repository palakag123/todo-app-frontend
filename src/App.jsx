import React, { useState } from "react";
import { BrowserRouter as Router, Routes, Route } from "react-router-dom";
import { CREATE_TODO_ROUTE, HOME_ROUTE } from "./constants/routes";
import CreateTodopage from "./pages/CreateTodoPage";
import TodoListPage from "./pages/TodoListPage";

const dummyTodos = [{ id: 1, todo: "first todo list item" }];
function App() {
  const [todos, setTodos] = useState(dummyTodos);

  return (
    <Router>
      <Routes>
        <Route path={HOME_ROUTE} element={<TodoListPage todos={todos} />} />
        <Route
          path={CREATE_TODO_ROUTE}
          element={<CreateTodopage todos={todos} setTodos={setTodos} />}
        />
      </Routes>
    </Router>
  );
}

export default App;
