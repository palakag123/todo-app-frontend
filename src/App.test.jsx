import React from "react";
import { render } from "@testing-library/react";
import App from "./App";

describe("app", () => {
  it("should take a snapshot for App", () => {
    const { asFragment } = render(<App />);
    expect(asFragment(<App />)).toMatchSnapshot();
  });
});
