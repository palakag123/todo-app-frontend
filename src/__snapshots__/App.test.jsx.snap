// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app should take a snapshot for App 1`] = `
<DocumentFragment>
  <div
    class="todo-list-page-container"
  >
    <div
      class="details-container"
    >
      <div
        class="details-inner-container"
      >
        <img
          alt="user"
          data-testid="userImg"
          id="user-img"
          src="userImage.jpeg"
        />
        <h2
          data-testid="greeting"
          id="greeting"
        >
          HI, USER
        </h2>
        <br />
        <h1
          data-testid="time"
          id="date-time"
        >
          <time
            datetime="1650831384895"
          >
            1:46 am
          </time>
        </h1>
        <h1
          data-testid="date"
          id="date-time"
        >
          <time
            datetime="1650831384898"
          >
            25 April, 2022
          </time>
        </h1>
      </div>
    </div>
    <div
      class="allTodos-container"
    >
      <h1
        data-testid="allTodoHeading"
        id="todos-heading"
      >
        ALL TO-DOS
      </h1>
      <div
        class="all-cards-outer-container"
        data-testid="allCards"
      >
        <div
          class="all-cards-container"
          data-testid="allCardsInner"
        >
          <div
            class="todo-card"
            data-testid="todoCard"
          >
            <p
              id="todo-title"
            >
              first todo list item
            </p>
          </div>
        </div>
      </div>
      <button
        class="new-todo-button"
        data-testid="newTodoButton"
        type="button"
      >
        CREATE NEW
      </button>
    </div>
  </div>
</DocumentFragment>
`;
